Cannot compile

Moonshine cannot compile as of today morning for me. Something todo with referencing fields in the enet bindings that don't exist. I initially thought this was a problem with my system since none of the commits since my last successful build made any significant changes to affect this. Then I realized that something else is up when I saw that the same error is happening in the clippy github action in this repo. 

I've included a copy of those logs below: (apologies for the formatting issues and extra characters. Im on my phone atm)

```
2026-03-14T22:08:11.7587634Z [1m[92m    Checking[0m enet v0.3.0
2026-03-14T22:08:11.8382673Z [1m[91merror[E0609][0m[1m: no field `peerCount` on type `_ENetHost`[0m
2026-03-14T22:08:11.8385944Z    [1m[94m--> [0m/github/home/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/enet-0.3.0/src/host.rs:140:49
2026-03-14T22:08:11.8387049Z     [1m[94m|[0m
2026-03-14T22:08:11.8387962Z [1m[94m140[0m [1m[94m|[0m         let peer_count = unsafe { (*self.inner).peerCount.try_into().expect("too many peers") };
2026-03-14T22:08:11.8389217Z     [1m[94m|[0m                                                 [1m[91m^^^^^^^^^[0m [1m[91munknown field[0m
2026-03-14T22:08:11.8389870Z     [1m[94m|[0m
2026-03-14T22:08:11.8390342Z     [1m[94m= [0m[1mnote[0m: available field is: `_address`
2026-03-14T22:08:11.8390730Z 
2026-03-14T22:08:11.8399709Z [1m[91merror[E0609][0m[1m: no field `peers` on type `_ENetHost`[0m
2026-03-14T22:08:11.8401125Z    [1m[94m--> [0m/github/home/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/enet-0.3.0/src/host.rs:142:79
2026-03-14T22:08:11.8402235Z     [1m[94m|[0m
2026-03-14T22:08:11.8403894Z [1m[94m142[0m [1m[94m|[0m         let raw_peers = unsafe { std::slice::from_raw_parts_mut((*self.inner).peers, peer_count) };
2026-03-14T22:08:11.8405370Z     [1m[94m|[0m                                                                               [1m[91m^^^^^[0m [1m[91munknown field[0m
2026-03-14T22:08:11.8406280Z     [1m[94m|[0m
2026-03-14T22:08:11.8407517Z     [1m[94m= [0m[1mnote[0m: available field is: `_address`
2026-03-14T22:08:11.8412514Z 
2026-03-14T22:08:11.8465489Z [1m[91merror[E0609][0m[1m: no field `type_` on type `&'b _ENetEvent`[0m
2026-03-14T22:08:11.8467197Z   [1m[94m--> [0m/github/home/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/enet-0.3.0/src/event.rs:36:25
2026-03-14T22:08:11.8468360Z    [1m[94m|[0m
2026-03-14T22:08:11.8469043Z [1m[94m36[0m [1m[94m|[0m         match event_sys.type_ {
2026-03-14T22:08:11.8469964Z    [1m[94m|[0m                         [1m[91m^^^^^[0m [1m[91munknown field[0m
2026-03-14T22:08:11.8470749Z    [1m[94m|[0m
2026-03-14T22:08:11.8471430Z    [1m[94m= [0m[1mnote[0m: available field is: `_address`
2026-03-14T22:08:11.8475926Z 
2026-03-14T22:08:11.8483079Z [1m[91merror[E0609][0m[1m: no field `peer` on type `&'b _ENetEvent`[0m
2026-03-14T22:08:11.8484805Z   [1m[94m--> [0m/github/home/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/enet-0.3.0/src/event.rs:39:57
2026-03-14T22:08:11.8490038Z    [1m[94m|[0m
2026-03-14T22:08:11.8491311Z [1m[94m39[0m [1m[94m|[0m                 Some(Event::Connect(Peer::new(event_sys.peer)))
2026-03-14T22:08:11.8492415Z    [1m[94m|[0m                                                         [1m[91m^^^^[0m [1m[91munknown field[0m
2026-03-14T22:08:11.8493122Z    [1m[94m|[0m
2026-03-14T22:08:11.8493657Z    [1m[94m= [0m[1mnote[0m: available field is: `_address`
2026-03-14T22:08:11.8494087Z 
2026-03-14T22:08:11.8511516Z [1m[91merror[E0609][0m[1m: no field `peer` on type `&'b _ENetEvent`[0m
2026-03-14T22:08:11.8513025Z   [1m[94m--> [0m/github/home/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/enet-0.3.0/src/event.rs:42:60
2026-03-14T22:08:11.8514210Z    [1m[94m|[0m
2026-03-14T22:08:11.8515200Z [1m[94m42[0m [1m[94m|[0m                 Some(Event::Disconnect(Peer::new(event_sys.peer), event_sys.data))
2026-03-14T22:08:11.8516459Z    [1m[94m|[0m                                                            [1m[91m^^^^[0m [1m[91munknown field[0m
2026-03-14T22:08:11.8517492Z    [1m[94m|[0m
2026-03-14T22:08:11.8518193Z    [1m[94m= [0m[1mnote[0m: available field is: `_address`
2026-03-14T22:08:11.8518603Z 
2026-03-14T22:08:11.8519151Z [1m[91merror[E0609][0m[1m: no field `data` on type `&'b _ENetEvent`[0m
2026-03-14T22:08:11.8520337Z   [1m[94m--> [0m/github/home/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/enet-0.3.0/src/event.rs:42:77
2026-03-14T22:08:11.8521193Z    [1m[94m|[0m
2026-03-14T22:08:11.8522061Z [1m[94m42[0m [1m[94m|[0m                 Some(Event::Disconnect(Peer::new(event_sys.peer), event_sys.data))
2026-03-14T22:08:11.8523183Z    [1m[94m|[0m                                                                             [1m[91m^^^^[0m [1m[91munknown field[0m
2026-03-14T22:08:11.8523860Z    [1m[94m|[0m
2026-03-14T22:08:11.8524334Z    [1m[94m= [0m[1mnote[0m: available field is: `_address`
2026-03-14T22:08:11.8524718Z 
2026-03-14T22:08:11.8525106Z [1m[91merror[E0609][0m[1m: no field `peer` on type `&'b _ENetEvent`[0m
2026-03-14T22:08:11.8526240Z   [1m[94m--> [0m/github/home/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/enet-0.3.0/src/event.rs:45:45
2026-03-14T22:08:11.8527304Z    [1m[94m|[0m
2026-03-14T22:08:11.8527935Z [1m[94m45[0m [1m[94m|[0m                 sender: Peer::new(event_sys.peer),
2026-03-14T22:08:11.8528781Z    [1m[94m|[0m                                             [1m[91m^^^^[0m [1m[91munknown field[0m
2026-03-14T22:08:11.8529376Z    [1m[94m|[0m
2026-03-14T22:08:11.8529846Z    [1m[94m= [0m[1mnote[0m: available field is: `_address`
2026-03-14T22:08:11.8530211Z 
2026-03-14T22:08:11.8530632Z [1m[91merror[E0609][0m[1m: no field `channelID` on type `&'b _ENetEvent`[0m
2026-03-14T22:08:11.8531778Z   [1m[94m--> [0m/github/home/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/enet-0.3.0/src/event.rs:46:39
2026-03-14T22:08:11.8532630Z    [1m[94m|[0m
2026-03-14T22:08:11.8533461Z [1m[94m46[0m [1m[94m|[0m                 channel_id: event_sys.channelID,
2026-03-14T22:08:11.8534343Z    [1m[94m|[0m                                       [1m[91m^^^^^^^^^[0m [1m[91munknown field[0m
2026-03-14T22:08:11.8534965Z    [1m[94m|[0m
2026-03-14T22:08:11.8535445Z    [1m[94m= [0m[1mnote[0m: available field is: `_address`
2026-03-14T22:08:11.8535814Z 
2026-03-14T22:08:11.8536216Z [1m[91merror[E0609][0m[1m: no field `packet` on type `&'b _ENetEvent`[0m
2026-03-14T22:08:11.8537499Z   [1m[94m--> [0m/github/home/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/enet-0.3.0/src/event.rs:47:59
2026-03-14T22:08:11.8538356Z    [1m[94m|[0m
2026-03-14T22:08:11.8539147Z [1m[94m47[0m [1m[94m|[0m                 packet: Packet::from_sys_packet(event_sys.packet),
2026-03-14T22:08:11.8540204Z    [1m[94m|[0m                                                           [1m[91m^^^^^^[0m [1m[91munknown field[0m
2026-03-14T22:08:11.8540884Z    [1m[94m|[0m
2026-03-14T22:08:11.8541388Z    [1m[94m= [0m[1mnote[0m: available field is: `_address`
2026-03-14T22:08:11.8542056Z 
2026-03-14T22:08:11.8542519Z [1m[91merror[E0609][0m[1m: no field `type_` on type `&'b _ENetEvent`[0m
2026-03-14T22:08:11.8543722Z   [1m[94m--> [0m/github/home/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/enet-0.3.0/src/event.rs:49:66
2026-03-14T22:08:11.8544661Z    [1m[94m|[0m
2026-03-14T22:08:11.8545404Z [1m[94m49[0m [1m[94m|[0m             _ => panic!("unrecognized event type: {}", event_sys.type_),
2026-03-14T22:08:11.8546496Z    [1m[94m|[0m                                                                  [1m[91m^^^^^[0m [1m[91munknown field[0m
2026-03-14T22:08:11.8547426Z    [1m[94m|[0m
2026-03-14T22:08:11.8547966Z    [1m[94m= [0m[1mnote[0m: available field is: `_address`
2026-03-14T22:08:11.8548382Z 
2026-03-14T22:08:11.8554891Z [1m[91merror[E0609][0m[1m: no field `channelLimit` on type `_ENetHost`[0m
2026-03-14T22:08:11.8556117Z    [1m[94m--> [0m/github/home/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/enet-0.3.0/src/host.rs:113:60
2026-03-14T22:08:11.8557150Z     [1m[94m|[0m
2026-03-14T22:08:11.8558024Z [1m[94m113[0m [1m[94m|[0m         ChannelLimit::from_enet_val(unsafe { (*self.inner).channelLimit })
2026-03-14T22:08:11.8559122Z     [1m[94m|[0m                                                            [1m[91m^^^^^^^^^^^^[0m [1m[91munknown field[0m
2026-03-14T22:08:11.8559800Z     [1m[94m|[0m
2026-03-14T22:08:11.8560285Z     [1m[94m= [0m[1mnote[0m: available field is: `_address`
2026-03-14T22:08:11.8560685Z 
2026-03-14T22:08:11.8561124Z [1m[91merror[E0609][0m[1m: no field `incomingBandwidth` on type `_ENetHost`[0m
2026-03-14T22:08:11.8562292Z    [1m[94m--> [0m/github/home/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/enet-0.3.0/src/host.rs:118:32
2026-03-14T22:08:11.8563148Z     [1m[94m|[0m
2026-03-14T22:08:11.8563713Z [1m[94m118[0m [1m[94m|[0m         unsafe { (*self.inner).incomingBandwidth }
2026-03-14T22:08:11.8564579Z     [1m[94m|[0m                                [1m[91m^^^^^^^^^^^^^^^^^[0m [1m[91munknown field[0m
2026-03-14T22:08:11.8565190Z     [1m[94m|[0m
2026-03-14T22:08:11.8565647Z     [1m[94m= [0m[1mnote[0m: available field is: `_address`
2026-03-14T22:08:11.8566025Z 
2026-03-14T22:08:11.8566450Z [1m[91merror[E0609][0m[1m: no field `outgoingBandwidth` on type `_ENetHost`[0m
2026-03-14T22:08:11.8567757Z    [1m[94m--> [0m/github/home/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/enet-0.3.0/src/host.rs:123:32
2026-03-14T22:08:11.8568598Z     [1m[94m|[0m
2026-03-14T22:08:11.8569151Z [1m[94m123[0m [1m[94m|[0m         unsafe { (*self.inner).outgoingBandwidth }
2026-03-14T22:08:11.8569994Z     [1m[94m|[0m                                [1m[91m^^^^^^^^^^^^^^^^^[0m [1m[91munknown field[0m
2026-03-14T22:08:11.8570589Z     [1m[94m|[0m
2026-03-14T22:08:11.8571045Z     [1m[94m= [0m[1mnote[0m: available field is: `_address`
2026-03-14T22:08:11.8571729Z 
2026-03-14T22:08:11.8572158Z [1m[91merror[E0609][0m[1m: no field `address` on type `_ENetHost`[0m
2026-03-14T22:08:11.8573358Z    [1m[94m--> [0m/github/home/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/enet-0.3.0/src/host.rs:128:60
2026-03-14T22:08:11.8574263Z     [1m[94m|[0m
2026-03-14T22:08:11.8575091Z [1m[94m128[0m [1m[94m|[0m         Address::from_enet_address(&unsafe { (*self.inner).address })
2026-03-14T22:08:11.8576171Z     [1m[94m|[0m                                                            [1m[91m^^^^^^^[0m [1m[91munknown field[0m
2026-03-14T22:08:11.8577031Z     [1m[94m|[0m
2026-03-14T22:08:11.8577509Z [1m[96mhelp[0m: a field with a similar name exists
2026-03-14T22:08:11.8578011Z     [1m[94m|[0m
2026-03-14T22:08:11.8578878Z [1m[94m128[0m [1m[94m| [0m        Address::from_enet_address(&unsafe { (*self.inner).[92m_[0maddress })
2026-03-14T22:08:11.8579844Z     [1m[94m|[0m                                                            [92m+[0m
2026-03-14T22:08:11.8580306Z 
2026-03-14T22:08:11.8580724Z [1m[91merror[E0609][0m[1m: no field `peerCount` on type `_ENetHost`[0m
2026-03-14T22:08:11.8582196Z    [1m[94m--> [0m/github/home/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/enet-0.3.0/src/host.rs:133:32
2026-03-14T22:08:11.8583049Z     [1m[94m|[0m
2026-03-14T22:08:11.8583584Z [1m[94m133[0m [1m[94m|[0m         unsafe { (*self.inner).peerCount }
2026-03-14T22:08:11.8584390Z     [1m[94m|[0m                                [1m[91m^^^^^^^^^[0m [1m[91munknown field[0m
2026-03-14T22:08:11.8584978Z     [1m[94m|[0m
2026-03-14T22:08:11.8585440Z     [1m[94m= [0m[1mnote[0m: available field is: `_address`
2026-03-14T22:08:11.8585819Z 
2026-03-14T22:08:11.8881460Z [1mFor more information about this error, try `rustc --explain E0609`.[0m
2026-03-14T22:08:11.8921887Z [1m[91merror[0m: could not compile `enet` (lib) due to 15 previous errors
2026-03-14T22:08:11.8922880Z [1m[33mwarning[0m: build failed, waiting for other jobs to finish...
2026-03-14T22:08:17.3939201Z ##[error]Process completed with exit code 101.
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -65,3 +65,4 @@ zeroconf = "0.17.0"
 
 [patch.crates-io]
 reed-solomon-erasure = { version = "6.0.0", git = "https://github.com/hgaiser/reed-solomon-erasure", branch = "moonshine" }
+enet-sys = { git = "https://github.com/eslam-allam/enet-sys", branch = "master" }

